components/board/fake/board.go

```diff
@@ -21,6 +22,8 @@ import (
 	"go.viam.com/rdk/resource"
 )
 
+var randomPin = "4"
+
```

Comment:
okay I don't love this but returning a random number broke a lot of tests that use fake analogs. my logic here is that the test will all behave as normal, but if a user wants to see random numbers on the RC card they have to use a specific pin, in this case 4 (but we can change that, it was a random choice)

Reply:
Put a comment on this. From the name alone, I can't tell if this is supposed to be only for testing, or for production, or what it would do even if I used it (maybe it's like https://xkcd.com/221/). I like the Github comment you wrote; just put that in the code.

Reply:
actually realized it probably makes more sense to do it this way, only pin 0 will always output 0

Reply:
just came back to this and had some major issues with `TestRobotReconfigure`, so the logic for analogs and digitals is a bit conflicting. for analogs any pin will report incrementing non-zero values except pin 0. for digitals, _only_ pin 0 will report incrementing non-zero values.